jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/javacore/parser/j9/section/memory/MemoryTagParser.java

```diff
@@ -74,5 +77,42 @@ public void processLine(String source, int startingOffset) {
 		};
 		addTag(T_1STSEGMENT, lineRule);
 	}
+
+	protected void addHeapSpace() {
+		// NULL           id                 start              end                size               space/region
+		// 1STHEAPSPACE   0x0000019EEF4B3E40         --                 --                 --         Generational
+		// 1STHEAPREGION  0x0000019EEF53FF60 0x00000007224F0000 0x0000000722E70000 0x0000000000980000 Generational/Tenured Region
+		ILineRule lineRule = new LineRule() {
+			public void processLine(String source, int startingOffset) {
+				addPrefixedHexToken(MEMORY_SEGMENT_ID);
+				consumeUntilFirstMatch(CommonPatternMatchers.whitespace);
+				consumeUntilFirstMatch(CommonPatternMatchers.non_whitespace_attribute);
+				consumeUntilFirstMatch(CommonPatternMatchers.whitespace);
+				consumeUntilFirstMatch(CommonPatternMatchers.non_whitespace_attribute);
+				consumeUntilFirstMatch(CommonPatternMatchers.whitespace);
+				consumeUntilFirstMatch(CommonPatternMatchers.non_whitespace_attribute);
+				consumeUntilFirstMatch(CommonPatternMatchers.whitespace);
```

Comment:
If the number of groups changes then it might get the wrong name for the heap space. It is not worth trying to be more general as the Javacore format doesn't often change, and if it does I have no idea what it would change to.
We would then just update the Javacore parser - as generally the later parsers understand all Javacores, but
an old parser might not understand Javacore files in a changed format.

Reply:
Rather than ignore start/end/size, perhaps this should collect those pieces of information.